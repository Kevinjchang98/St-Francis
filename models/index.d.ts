export declare type Client = {
    readonly id: string;
    readonly notes?: string | null;
    readonly gender?: string | null;
    readonly numKids?: string | null;
    readonly birthday?: string | null;
    readonly isBanned?: string | null;
    readonly lastName?: string | null;
    readonly firstName?: string | null;
    readonly lastNameLower?: string | null;
    readonly firstNameLower?: string | null;
    readonly createdAt?: string | null;
    readonly lastUpdate?: string | null;
    readonly postalCode?: string | null;
    readonly isCheckedIn?: string | null;
    readonly numInFamily?: string | null;
    readonly lastBackpack?: string | null;
    readonly lastBusTicket?: string | null;
    readonly middleInitial?: string | null;
    readonly lastSleepingbag?: string | null;
    readonly BerthaPittsCampbellResident?: string | null;
    readonly lastFinancialAssistanceDate?: string | null;
    readonly lastFinancialAssistanceValue?: string | null;
};

export declare type Visit = {
    readonly id: string;
    readonly kids?: string | null;
    readonly mens?: string | null;
    readonly soap?: string | null;
    readonly kidsQ?: string | null;
    readonly mensQ?: string | null;
    readonly notes?: string | null;
    readonly oldID?: string | null;
    readonly razor?: string | null;
    readonly boyAge?: string | null;
    readonly diaper?: string | null;
    readonly lotion?: string | null;
    readonly womens?: string | null;
    readonly girlAge?: string | null;
    readonly womensQ?: string | null;
    readonly backpack?: string | null;
    readonly giftCard?: string | null;
    readonly busTicket?: string | null;
    readonly household?: string | null;
    readonly timestamp?: string | null;
    readonly toothbrush?: string | null;
    readonly toothpaste?: string | null;
    readonly clothingBoy?: string | null;
    readonly clothingMen?: string | null;
    readonly sleepingBag?: string | null;
    readonly clothingGirl?: string | null;
    readonly lastBackpack?: string | null;
    readonly clothingWomen?: string | null;
    readonly lastSleepingBag?: string | null;
    readonly shampooConditioner?: string | null;
    readonly financialAssistance?: string | null;
    readonly lastFinancialAssistance?: string | null;
};
