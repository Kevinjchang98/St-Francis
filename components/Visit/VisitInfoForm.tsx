'use client';

import { Button, Form, FormItem, FormRow } from '@/components/UI';
import { Visit } from '@/models/index';
import { useState } from 'react';
import styles from './VisitInfoForm.module.css';

interface VisitInfoFormProps {
    initialVisitData?: Visit;
    onSubmit?: (visitData: Visit) => void;
    submitLabel?: string;
    onChange?: (visitData: Visit) => void;
}

const defaultVisitData = {
    id: '',
    clothingMen: false,
    clothingWomen: false,
    clothingBoy: false,
    clothingGirl: false,
    household: '',
    notes: '',
    backpack: false,
    sleepingBag: false,
    food: false,
    mensQ: '',
    womensQ: '',
    kidsQ: '',
    householdItem: false,
};

const toInt = (value: string) => parseInt(value) || 0;
const toString = (value: number | undefined | null) => value?.toString() || '';

export default function VisitInfoForm({
    onSubmit,
    initialVisitData,
    submitLabel,
    onChange,
}: VisitInfoFormProps) {
    // Append boyAge and girlAge to notes
    let initialNotes = initialVisitData?.notes || '';
    initialVisitData?.boyAge &&
        (initialNotes += `\nBoy Age: ${initialVisitData?.boyAge} `);
    initialVisitData?.girlAge &&
        (initialNotes += `\nGirl Age: ${initialVisitData?.girlAge} `);

    const [visitData, setVisitData] = useState({
        ...defaultVisitData,
        ...initialVisitData,
        busTicket: toString(initialVisitData?.busTicket),
        orcaCard: toString(initialVisitData?.orcaCard),
        giftCard: toString(initialVisitData?.giftCard),
        diaper: toString(initialVisitData?.diaper),
        financialAssistance: toString(initialVisitData?.financialAssistance),
        mensQ: toString(initialVisitData?.mensQ),
        womensQ: toString(initialVisitData?.womensQ),
        kidsQ: toString(initialVisitData?.kidsQ),
        notes: initialNotes,
    });

    const handleSubmit = async (e: any) => {
        e.preventDefault(); // Prevent redirect
        onSubmit && onSubmit(transformData(visitData));
    };

    const transformData = (visitData: any) => {
        return {
            ...visitData,
            busTicket: toInt(visitData.busTicket),
            orcaCard: toInt(visitData.orcaCard),
            giftCard: toInt(visitData.giftCard),
            diaper: toInt(visitData.diaper),
            financialAssistance: toInt(visitData.financialAssistance),
            // default to 0 if unchecked before submit
            mensQ: visitData.clothingMen ? toInt(visitData.mensQ) : 0,
            womensQ: visitData.clothingWomen ? toInt(visitData.womensQ) : 0,
            kidsQ:
                visitData.clothingBoy || visitData.clothingGirl
                    ? toInt(visitData.kidsQ)
                    : 0,
            // transfer boyAge and girlAge to notes with submit
            boyAge: '',
            girlAge: '',
        };
    };

    const handleChange = (key: string) => (e: any) => {
        let value = '';
        if (e.target.type === 'checkbox') {
            value = e.target.checked;
        } else value = e.target.value;

        if (key === 'householdItem') {
            const booleanValue = Boolean(value);

            setVisitData((visitData) => ({
                ...visitData,
                // reset household items notes if householdItem is unchecked
                household: !booleanValue ? '' : visitData.household,
                householdItem: booleanValue,
            }));
        } else {
            setVisitData((visitData) => ({
                ...visitData,
                [key]: value,
            }));
        }

        if (onChange) {
            const updatedData = {
                ...visitData,
                [key]: value,
                ...(key === 'householdItem' && !value && { household: '' }),
            };
            onChange(transformData(updatedData));
        }
    };
    return (
        <Form onSubmit={handleSubmit}>
            <h2>Clothing</h2>

            <FormRow className={styles.formRows}>
                <FormRow className={styles.rowItems}>
                    <FormItem
                        label="Men"
                        type="checkbox"
                        id="clothingMen"
                        checked={!!visitData.clothingMen}
                        onChange={handleChange('clothingMen')}
                    />
                    <FormItem
                        type="number"
                        id="mensQ"
                        value={visitData.mensQ}
                        onChange={handleChange('mensQ')}
                        hidden={!visitData.clothingMen}
                    />
                </FormRow>
                <FormRow className={styles.rowItems}>
                    <FormItem
                        type="checkbox"
                        id="clothingWomen"
                        label="Women"
                        checked={!!visitData.clothingWomen}
                        onChange={handleChange('clothingWomen')}
                    />
                    <FormItem
                        type="number"
                        id="womensQ"
                        value={visitData.womensQ}
                        onChange={handleChange('womensQ')}
                        hidden={!visitData.clothingWomen}
                    />
                </FormRow>
                <FormRow className={styles.rowItems}>
                    <FormItem
                        type="checkbox"
                        id="clothingBoy"
                        label="Kids (Boy)"
                        checked={!!visitData.clothingBoy}
                        onChange={handleChange('clothingBoy')}
                    />
                    <FormItem
                        type="checkbox"
                        id="clothingGirl"
                        label="Kids (Girl)"
                        checked={!!visitData.clothingGirl}
                        onChange={handleChange('clothingGirl')}
                    />
                    <FormItem
                        type="number"
                        id="kidsQ"
                        value={visitData.kidsQ}
                        onChange={handleChange('kidsQ')}
                        hidden={
                            !visitData.clothingBoy && !visitData.clothingGirl
                        }
                    />
                </FormRow>
            </FormRow>

            <h2>Special Requests</h2>

            <FormRow>
                <FormItem
                    type="number"
                    id="busTicket"
                    label="Bus Ticket"
                    value={visitData.busTicket}
                    onChange={handleChange('busTicket')}
                />
                <FormItem
                    type="number"
                    id="orcaCard"
                    label="Orca Card"
                    value={visitData.orcaCard}
                    onChange={handleChange('orcaCard')}
                />
                <FormItem
                    type="number"
                    id="giftCard"
                    label="Gift Card"
                    value={visitData.giftCard}
                    onChange={handleChange('giftCard')}
                />
                <FormItem
                    type="number"
                    id="diaper"
                    label="Diaper"
                    value={visitData.diaper}
                    onChange={handleChange('diaper')}
                />
                <FormItem
                    type="number"
                    id="financialAssistance"
                    label="Financial Assistance"
                    value={visitData.financialAssistance}
                    onChange={handleChange('financialAssistance')}
                />
            </FormRow>
            <FormRow>
                <FormItem
                    type="checkbox"
                    id="backpack"
                    label="Backpack"
                    checked={!!visitData.backpack}
                    onChange={handleChange('backpack')}
                />
                <FormItem
                    type="checkbox"
                    id="sleepingBag"
                    label="Sleeping Bag"
                    checked={!!visitData.sleepingBag}
                    onChange={handleChange('sleepingBag')}
                />
                <FormItem
                    type="checkbox"
                    id="food"
                    label="Food"
                    checked={!!visitData.food}
                    onChange={handleChange('food')}
                />
                <FormItem
                    type="checkbox"
                    id="householdItem"
                    label="Household Items"
                    checked={!!visitData.householdItem}
                    onChange={handleChange('householdItem')}
                />
            </FormRow>

            <div hidden={!visitData.householdItem}>
                <h2>Household items</h2>
                <FormItem
                    type="textarea"
                    id="household"
                    rows={3}
                    value={visitData.household || ''}
                    onChange={handleChange('household')}
                    // hidden={!visitData.householdItem}
                />
            </div>

            <h2>Notes</h2>

            <FormItem
                type="textarea"
                id="notes"
                rows={3}
                value={visitData.notes || ''}
                onChange={handleChange('notes')}
            />
            <Button type="submit" className={styles.submitBtn}>
                {submitLabel || 'Submit'}
            </Button>
        </Form>
    );
}
