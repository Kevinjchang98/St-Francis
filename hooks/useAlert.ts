import { AlertContext } from '@/providers/AlertProvider';
import { useContext } from 'react';

export const useAlert = () => {
    const { alert, setAlert } = useContext(AlertContext);
    return [alert, setAlert] as const;
};
export default useAlert;
