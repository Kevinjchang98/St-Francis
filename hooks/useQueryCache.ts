import { CLIENTS_PATH, VISITS_PATH } from '@/utils/queries';
import { useQueryClient } from 'react-query';
import { Client } from '../models';
import { DocFilter } from '../utils';
import { queryClient } from '@/providers/index';

export default function useQueryCache() {
    // TODO: update cached query data instead of removing it, no refetch needed
    // TODO: move this logic to the updateClient helper function
    const updateClientCache = (clientData: DocFilter) => {
        // Remove cached query data when update is successful, refetch on next page load
        queryClient.removeQueries([CLIENTS_PATH, clientData.id]);
        // Remove cached query data for checked-in clients query, refetch on next page load
        queryClient.removeQueries([CLIENTS_PATH, 'checkedin']);

        // Update cached query data for searched clients query, no refetch needed
        queryClient.setQueriesData([CLIENTS_PATH, 'searched'], (data: any) => {
            if (!data) return;
            data['clients']?.forEach((client: Client, index: number) => {
                if (client.id === clientData.id)
                    data['clients'][index] = { ...client, ...clientData };
            });
            return data;
        });
    };

    const updateVisitCache = (clientID: string, visitData?: DocFilter) => {
        queryClient.removeQueries([CLIENTS_PATH, clientID, VISITS_PATH]);
        if (!visitData) return;
        queryClient.removeQueries([
            CLIENTS_PATH,
            clientID,
            VISITS_PATH,
            visitData.id,
        ]);
    };

    return { updateClientCache, updateVisitCache };
}
