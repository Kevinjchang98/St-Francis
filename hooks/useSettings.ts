import { SettingsContext } from '@/providers/SettingsProvider';
import { useContext } from 'react';

export function useSettings() {
    return useContext(SettingsContext);
}
