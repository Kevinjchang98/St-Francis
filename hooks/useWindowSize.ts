import { useState, useEffect } from 'react';

export const MOBILE_WIDTH = 768;

export type UseWindowSize = {
    isMobile: boolean;
    width: number;
    height: number;
};
/**
 *  Hook to get the window size width and height and if it is mobile
 */
export function useWindowSize(): UseWindowSize {
    const [size, setSize] = useState({ width: MOBILE_WIDTH, height: 0 });
    useEffect(() => {
        const resize = () =>
            setSize({
                width: window.innerWidth,
                height: window.innerHeight,
            });
        window.addEventListener('resize', resize);
        return () => window.removeEventListener('resize', resize);
    }, []);

    return { ...size, isMobile: size.width < MOBILE_WIDTH };
}
