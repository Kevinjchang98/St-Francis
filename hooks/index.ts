import { useAuth } from './useAuth';
import { useSettings } from './useSettings';

export { useAuth, useSettings };
