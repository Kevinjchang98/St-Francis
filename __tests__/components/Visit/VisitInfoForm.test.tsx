import VisitInfoForm from '@/components/Visit/VisitInfoForm';
import { Client } from '@/models/index';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { Timestamp } from 'firebase/firestore';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { useSettings } from '@/hooks/index';

jest.mock('@/hooks/index', () => ({
    useSettings: jest.fn(() => ({
        settings: {
            visitThreshold: 30,
            backpackThreshold: 180,
            sleepingbagThreshold: 180,
        },
    })),
}));

describe('Visit Info Form Component', () => {
    const epochTimestamp = Timestamp.fromDate(new Date(0));
    const mockClient: Client = {
        id: '47636',
        firstName: 'First',
        lastName: 'Last',
        race: 'Race',
        notes: 'Notes',
        gender: 'Gender',
        isCheckedIn: false,
        isBanned: false,
        numKids: 0,
        numInFamily: 0,
        postalCode: '0',
        lastBusTicket: epochTimestamp,
        lastSleepingbag: epochTimestamp,
        birthday: epochTimestamp,
        createdAt: epochTimestamp,
        updatedAt: epochTimestamp,
        lastBackpack: epochTimestamp,
    };
    const earlyClient = {
        ...mockClient,
        lastBusTicket: Timestamp.fromDate(new Date()),
        lastSleepingbag: Timestamp.fromDate(new Date()),
        lastBackpack: Timestamp.fromDate(new Date()),
        lastVisit: Timestamp.fromDate(new Date()),
    };

    const getInputField = (label: string) => {
        return (screen.getByLabelText(label) as HTMLInputElement).value;
    };

    it('matches snapshot without any props', async () => {
        const tree = renderer.create(<VisitInfoForm />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly without initial data', async () => {
        render(<VisitInfoForm />);
        expect(screen.queryByText('Men')).toBeInTheDocument();
        expect(screen.queryByText('Women')).toBeInTheDocument();
        expect(screen.queryByText('Kids (Boy)')).toBeInTheDocument();
        expect(screen.queryByText('Kids (Girl)')).toBeInTheDocument();
        expect(screen.queryByText('Backpack')).toBeInTheDocument();
        expect(screen.queryByText('Sleeping Bag')).toBeInTheDocument();
        expect(screen.queryByText('Bus Ticket')).toBeInTheDocument();
        expect(screen.queryByText('Gift Card')).toBeInTheDocument();
        expect(screen.queryByText('Diaper')).toBeInTheDocument();
        expect(screen.queryByText('Financial Assistance')).toBeInTheDocument();
    });

    it('renders correctly with initial Visit Data', async () => {
        render(<VisitInfoForm clientData={mockClient} />);
        expect(getInputField('Men')).toBe('off');
        expect(getInputField('Women')).toBe('off');
        expect(getInputField('Kids (Boy)')).toBe('off');
        expect(getInputField('Kids (Girl)')).toBe('off');
        expect(getInputField('Backpack')).toBe('off');
        expect(getInputField('Sleeping Bag')).toBe('off');
        expect(getInputField('Bus Ticket')).toBe('');
        expect(getInputField('Gift Card')).toBe('');
        expect(getInputField('Diaper')).toBe('');
        expect(getInputField('Financial Assistance')).toBe('');
    });

    it('saves correct output with mock onSubmit callback', async () => {
        const mockOnSave = jest.fn((VisitData) => VisitData);
        render(<VisitInfoForm clientData={mockClient} onSubmit={mockOnSave} />);
        const saveButton = screen.getByRole('button', { name: 'Check in' });
        fireEvent.click(saveButton);
        expect(mockOnSave).toHaveBeenCalledTimes(1);
    });

    it('saves forcely if validates failed with mock onSubmit callback', async () => {
        const clientData = {
            ...mockClient,
            lastBusTicket: Timestamp.fromDate(new Date()),
            lastSleepingbag: Timestamp.fromDate(new Date()),
            lastBackpack: Timestamp.fromDate(new Date()),
            lastVisit: Timestamp.fromDate(new Date()),
        };
        const mockOnSave = jest.fn((VisitData) => VisitData);
        render(<VisitInfoForm clientData={clientData} onSubmit={mockOnSave} />);
        const saveButton = screen.getByRole('button', { name: 'Check in' });
        fireEvent.click(saveButton);
        expect(mockOnSave).toHaveBeenCalledTimes(0);
        const forceSaveButton = screen.getByRole('button', {
            name: 'Force Check-in',
        });
        fireEvent.click(forceSaveButton);
        expect(mockOnSave).toHaveBeenCalledTimes(1);
    });

    it('saves forcely if validates failed with mock onSubmit callback', async () => {
        const mockOnSave = jest.fn((VisitData) => VisitData);
        render(
            <VisitInfoForm clientData={earlyClient} onSubmit={mockOnSave} />
        );
        const saveButton = screen.getByRole('button', { name: 'Check in' });
        fireEvent.click(saveButton);
        expect(mockOnSave).toHaveBeenCalledTimes(0);
        const forceSaveButton = screen.getByRole('button', {
            name: 'Force Check-in',
        });
        fireEvent.click(forceSaveButton);
        expect(mockOnSave).toHaveBeenCalledTimes(1);
    });

    it('changes input value correctly', async () => {
        render(<VisitInfoForm clientData={mockClient} />);
        const input = screen.getByRole('textbox', {
            name: /household items/i,
        }) as HTMLInputElement;
        await userEvent.type(input, '123');
        expect(input.value).toBe('123');
    });

    it('changes input value correctly', async () => {
        render(<VisitInfoForm clientData={mockClient} />);
        const input = screen.getByRole('spinbutton', {
            name: /bus ticket/i,
        }) as HTMLInputElement;
        await userEvent.clear(input);
        const saveButton = screen.getByRole('button', { name: 'Check in' });
        fireEvent.click(saveButton);
    });

    it('allows early override to checkin', async () => {
        (useSettings as jest.Mock).mockReturnValue({
            settings: { earlyOverride: true },
        });
        const clientData = {
            ...mockClient,
            lastBusTicket: Timestamp.fromDate(new Date()),
            lastSleepingbag: Timestamp.fromDate(new Date()),
            lastBackpack: Timestamp.fromDate(new Date()),
            lastVisit: Timestamp.fromDate(new Date()),
        };
        const mockOnSave = jest.fn((VisitData) => VisitData);
        render(
            <VisitInfoForm clientData={earlyClient} onSubmit={mockOnSave} />
        );
        const saveButton = screen.getByRole('button', { name: 'Check in' });
        fireEvent.click(saveButton);
        expect(mockOnSave).toHaveBeenCalledTimes(1);
    });
});
