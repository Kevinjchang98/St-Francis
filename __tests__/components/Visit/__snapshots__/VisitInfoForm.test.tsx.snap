// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visit Info Form Component matches snapshot without any props 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <h2>
    Clothing
  </h2>
  <div
    className="formRows"
  >
    <div
      className="formRowItem"
    >
      <label
        htmlFor="clothingMen"
      >
        Men
      </label>
      <input
        id="clothingMen"
        name="clothingMen"
        onChange={[Function]}
        type="checkbox"
        value="off"
      />
    </div>
    <div
      className="formRowItem"
    >
      <label
        htmlFor="clothingWomen"
      >
        Women
      </label>
      <input
        id="clothingWomen"
        name="clothingWomen"
        onChange={[Function]}
        type="checkbox"
        value="off"
      />
    </div>
    <div
      className="formRowItem"
    >
      <label
        htmlFor="clothingBoy"
      >
        Kids (Boy)
      </label>
      <input
        id="clothingBoy"
        name="clothingBoy"
        onChange={[Function]}
        type="checkbox"
        value="off"
      />
    </div>
    <div
      className="formRowItem"
    >
      <label
        htmlFor="clothingGirl"
      >
        Kids (Girl)
      </label>
      <input
        id="clothingGirl"
        name="clothingGirl"
        onChange={[Function]}
        type="checkbox"
        value="off"
      />
    </div>
  </div>
  <h2>
    Special Requests
  </h2>
  <div
    className="formRows"
  >
    <div
      className="formRowItem"
    >
      <label
        htmlFor="backpack"
      >
        Backpack
      </label>
      <input
        id="backpack"
        name="backpack"
        onChange={[Function]}
        type="checkbox"
        value="off"
      />
    </div>
    <div
      className="formRowItem"
    >
      <label
        htmlFor="sleepingBag"
      >
        Sleeping Bag
      </label>
      <input
        id="sleepingBag"
        name="sleepingBag"
        onChange={[Function]}
        type="checkbox"
        value="off"
      />
    </div>
    <div
      className="formRowItem"
    >
      <label
        htmlFor="busTicket"
      >
        Bus Ticket
      </label>
      <input
        id="busTicket"
        name="busTicket"
        onChange={[Function]}
        type="number"
        value=""
      />
    </div>
    <div
      className="formRowItem"
    >
      <label
        htmlFor="giftCard"
      >
        Gift Card
      </label>
      <input
        id="giftCard"
        name="giftCard"
        onChange={[Function]}
        type="number"
        value=""
      />
    </div>
    <div
      className="formRowItem"
    >
      <label
        htmlFor="diaper"
      >
        Diaper
      </label>
      <input
        id="diaper"
        name="diaper"
        onChange={[Function]}
        type="number"
        value=""
      />
    </div>
    <div
      className="formRowItem"
    >
      <label
        htmlFor="financialAssistance"
      >
        Financial Assistance
      </label>
      <input
        id="financialAssistance"
        name="financialAssistance"
        onChange={[Function]}
        type="number"
        value=""
      />
    </div>
  </div>
  <label
    htmlFor="household"
  >
    <h2>
      Household items
    </h2>
  </label>
  <input
    id="household"
    name="household"
    onChange={[Function]}
    type="text"
    value=""
  />
  <label
    htmlFor="notes"
  >
    <h2>
      Notes
    </h2>
  </label>
  <input
    id="notes"
    name="notes"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Check in
  </button>
</form>
`;
